src/display/editor/tools.js

```diff
@@ -740,32 +738,41 @@ class AnnotationEditorUIManager {
     }
   }
 
+  /**
+   * Add an editor the current selection.
+   * @param {AnnotationEditor} editor
+   */
+  addToSelection(editor) {
+    if (this.#selectedEditors.has(editor)) {
+      this.#selectedEditors.delete(editor);
+      editor.unselect();
+      this.#dispatchUpdateStates({
+        hasSelectedEditor: this.hasSelection,
+      });
+      return;
+    }
+    this.#selectedEditors.add(editor);
+    editor.select();
+    this.#dispatchUpdateUI(editor.propertiesToUpdate);
+    this.#dispatchUpdateStates({
+      hasSelectedEditor: true,
+    });
+  }
+
   /**
    * Set the last selected editor.
    * @param {AnnotationEditor} editor
    */
   setSelected(editor) {
-    if (!this.#isMultipleSelection) {
-      if (this.#selectedEditors.has(editor)) {
-        if (this.#selectedEditors.size > 1) {
-          for (const ed of this.#selectedEditors) {
-            if (ed !== editor) {
-              ed.unselect();
-            }
-          }
-          this.#selectedEditors.clear();
-          this.#selectedEditors.add(editor);
-          this.#dispatchUpdateUI(editor.propertiesToUpdate);
-        }
-        return;
-      }
-
-      for (const ed of this.#selectedEditors) {
+    for (const ed of this.#selectedEditors) {
+      if (ed !== editor) {
         ed.unselect();
       }
-      this.#selectedEditors.clear();
     }
+    this.#selectedEditors.clear();
+
     this.#selectedEditors.add(editor);
+    editor.select();
     this.#dispatchUpdateUI(editor.propertiesToUpdate);
     this.#dispatchUpdateStates({
       hasSelectedEditor: this.hasSelection,
```

Comment:
This can probably be simplified to just the following, since the `add`-call above should guarantee that `this.hasSelection === true`:
```suggestion
      hasSelectedEditor: true,
```